File: lib/models/device/medsenser_device.dart
import 'package:medsenser_gate/models/device/measurement_node.dart';

class MedsenserDevice {
  final int deviceId;
  late String convertedDeviceId;
  final List<MeasurementNode> measurements = [];
  int batteryLevel = 0;
  DateTime lastSeen;
  late DateTime firstSeen;
  bool isNewDevice; // Flag indicating whether this is a new device

  MedsenserDevice({
    required this.deviceId, 
    required this.lastSeen,
    DateTime? firstSeen,
    this.isNewDevice = false
  }) {
    convertedDeviceId = "A4C35";
    this.firstSeen = firstSeen ?? lastSeen; // Use lastSeen as firstSeen if not specified
  }
}

File: ios/Runner/Model/MedsenserDevice.swift
//
//  MedsenserDevice.swift
//  Runner
//
//  Created by Mehmet Kadayıfçı on 19.02.2025.
//

// MedsenserDevice.swift

import Foundation

class MedsenserDevice {
    let deviceId: Int
    var convertedDeviceId: String
    var measurements: [MeasurementNode] = []
    var batteryLevel: Int = 0
    var lastSeen: Date?
    
    init(deviceId: Int) {
        self.deviceId = deviceId
        self.convertedDeviceId = "A4C35"
    }
}
